### PR TITLE
Add song selection with coconut mall

### PR DESCRIPTION
**Pull request by 2zqa**
Unsure if this fits the extension, but this pull requests adds a small popup menu with a drop-down menu containing an extra song. This is my first "serious" javascript code I ever wrote so feel free to review, roast and revise it.

What could be improved: using an enum/dictionary for song names instead of just passing strings. Wasn't sure how to implement that at the moment.

image

### CODE REVIEW

**Comment by corbindavenport** [comment]
This is good! I planned to add a track selector with more gaming shop themes (e.g. DSi Shop), I'll look into merging this (or at least using some of your code) once I get some free time.

File: js/background.js [hunk 1]
```diff
@@ -6,37 +6,69 @@ if ('mediaSession' in navigator) {
...
+        }
+    })
+}
+
+function setAudio(songName) {
+    currentSong = songName
+    // TODO: Fix creating new Audio objects every time you switch songs.
+    // Perhaps a dictionary/enum where strings are linked to Audio objects?
+    // Sadly I have only just finished my first year of IT without any js classes, only java, so unsure how 😅
+    themeAudio = new Audio(chrome.extension.getURL(songName + '.ogg'))
```
Comment (PandawanFr) [inline js/background.js:49]: To fix your issue of creating duplicate audio objects, you can simply use `themeAudio.src = chrome.extension.getURL(songName + '.ogg')`

File: popup/choose-song.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+<head>
+    <meta charset="utf-8">
+    <link rel="stylesheet" href="choose-song.css" />
+</head>
+
+<body>
+    <label for="songs">Choose a song:</label><br>
+
+    <select name="songs" id="songs">
+        <option value="wii-shop-theme" id="wii-shop-theme">Wii shop theme</option>
```
Comment (TheTechRobo) [inline popup/choose-song.html:13]: Try "Wii Shop Channel theme"

File: popup/choose-song.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        <option value="coconut-mall" id="coconut-mall">Coconut mall</option>
```
Comment (TheTechRobo) [inline popup/choose-song.html:14]: "Coconut Mall" (capital)

Sorry for being the grammar police :P

**Comment by corbindavenport** [comment]
I ended up going with a slightly different approach for this, because simply changing the audio element in the background doesn't make the media playback controls go away in Chrome (and other Chromium-based browsers). Chrome has no way to opt out of these media controls, so my only option is to set the Audio element `src` to be empty when the tab switches to a non-shopping site.

I did use your Coconut Mall theme though, thanks for that! This will be in the next update.

image

**Pull request closed without merging**